hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/MutableVolumeSet.java

```diff
@@ -267,14 +272,26 @@ void checkAllVolumes() throws IOException {
    * Handle one or more failed volumes.
    * @param failedVolumes
    */
-  private void handleVolumeFailures(Set<HddsVolume> failedVolumes) {
+  private void handleVolumeFailures(Set<HddsVolume> failedVolumes)
+      throws IOException {
     this.writeLock();
     try {
       for (HddsVolume v : failedVolumes) {
         // Immediately mark the volume as failed so it is unavailable
         // for new containers.
         failVolume(v.getHddsRootDir().getPath());
       }
+
+      // check failed volume tolerated
+      if (!hasEnoughVolumes()) {
+        // on startup, we could not try to stop uninitialized services
+        if (shutdownHook == null) {
+          throw new IOException("Don't have enough volumes to startup.");
```

Comment:
Prefer more detail error message, such as how may volumes configured,  and how many failed volumes.

Reply:
sure.